From 43dd73dfcdb137e4123674c216175766c9852920 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Fri, 6 Nov 2015 14:00:16 +0800
Subject: [PATCH 0655/5242] MLK-11829-2 ARM: dts: imx6sx-sabreauto: add egalax
 touch screen support

commit  afc451bcbb8f8589f73877af7b93803a1f7c8abd from
https://source.codeaurora.org/external/imx/linux-imx.git

Add egalax touch screen support on i2c2 bus.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index 6c8f44e..581fb1b 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -268,6 +268,16 @@
 		VLC-supply = <&reg_audio>;
 	};
 
+	egalax_ts@04 {
+		compatible = "eeti,egalax_ts";
+		reg = <0x04>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_egalax_int>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <22 2>;
+		wakeup-gpios = <&gpio6 22 GPIO_ACTIVE_HIGH>;
+	};
+
 	si4763: si4763@63 {
 		compatible = "si4761";
 		reg = <0x63>;
@@ -510,6 +520,12 @@
 		>;
 	};
 
+	pinctrl_egalax_int: egalax_intgrp {
+		fsl,pins = <
+			MX6SX_PAD_SD4_RESET_B__GPIO6_IO_22      0x80000000
+		>;
+	};
+
 	pinctrl_enet1_1: enet1grp-1 {
 		fsl,pins = <
 			MX6SX_PAD_ENET1_MDIO__ENET1_MDIO        0xa0b1
-- 
1.7.9.5

